marlowe-dashboard-client/src/Humanize.purs

```diff
@@ -66,11 +72,32 @@ humanizeInterval (SlotInterval from to) = humanize (formatSlot from) (formatSlot
     | fromDate == toDate = "on " <> fromDate <> " between " <> fromTime <> " and " <> toTime
     | otherwise = "between " <> fromDate <> " " <> fromTime <> " and " <> toDate <> " " <> toTime
 
-formatSlot :: Slot -> Maybe (Tuple String String)
-formatSlot slot = (\slotDT -> formatDate slotDT /\ formatTime slotDT) <$> slotToDateTime slot
-
-formatDate :: DateTime -> String
-formatDate =
+formatSlot :: Minutes -> Slot -> Maybe (Tuple String String)
+formatSlot tzOffset slot =
+  slotToDateTime slot
+    <#> \slotDT ->
+        let
+          adjustedDt = adjustTimeZone tzOffset slotDT
+        in
+          formatDate' adjustedDt /\ formatTime' adjustedDt
+
+-- This is a small wrapper around Now.getTimezoneOffset to be used in conjunction with
+-- adjustTimeZone. We need to negate the value in order to substract when we do the
+-- adjust, and Minutes does not have a Ring instance, so we need to unwrap and re-wrap
+getTimezoneOffset :: Effect Minutes
+getTimezoneOffset = wrap <<< negate <<< unwrap <$> Now.getTimezoneOffset
+
+-- Adjusts a DateTime via an offset (that can be obtained using getTimezoneOffset)
+-- The `adjust` function can overflow, if that happens (it shouldn't) we resolve to
+-- the original value
+adjustTimeZone :: Minutes -> DateTime -> DateTime
+adjustTimeZone tzOffset dt = fromMaybe dt (adjust tzOffset dt)
+
+formatDate :: Minutes -> DateTime -> String
+formatDate tzOffset dt = formatDate' $ adjustTimeZone tzOffset dt
+
+formatDate' :: DateTime -> String
+formatDate' =
```

Comment:
Another tiny thing, but the name of `formatDate'` (and `formatTime'`) doesn't convey what the function does. Can we have maybe `formatDate'` -> `formatDate`, and `formatDate` -> `formatDateWithOffset`? (Ditto for `formatTime`.)

Reply:
Well, the idea is to strongly encourage the developer to use the localized version. If you have `formatDate` and `formatDateWithOffset` in a view, and you don't currently have access to the `tzOffset`, you may be inclined to just use `formatDate` directly, which would cause confusion to the end-user. When we format a date is because we are showing it to the user, and when we do that we always want to show the localized version. The `formatDate'` is kind of a "private/internal" function that you should only use if you know what you are doing. I actually thought of not exporting it.

Reply:
Fair enough. In that case I wouldn't object to not exporting it. Or even putting it in a `where` clause in the `formatDate` function.